src/client/clientauth.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+Minetest
+Copyright (C) 2024 SFENCE, <sfence.software@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "clientauth.h"
+#include "util/auth.h"
+#include "util/srp.h"
+
+ClientAuth::ClientAuth() :
+		m_is_empty(true),
+		m_srp_verifier(""),
+		m_srp_salt("")
+{
+}
+
+ClientAuth::ClientAuth(const std::string &player_name, const std::string &password)
+{
+	applyPassword(player_name, password);
+}
+
+ClientAuth::~ClientAuth()
+{
+	deleteAuthData();
+}
+
+void ClientAuth::moveFrom(ClientAuth& other)
+{
+	m_is_empty = other.m_is_empty;
+	m_srp_verifier = other.m_srp_verifier;
+	m_srp_salt = other.m_srp_salt;
+
+	deleteAuthData();
+	
+	m_legacy_auth_data = other.m_legacy_auth_data;
+	m_srp_auth_data = other.m_srp_auth_data;
+	
+	other.m_legacy_auth_data = nullptr;
+	other.m_srp_auth_data = nullptr;
+}
+
+void ClientAuth::applyPassword(const std::string &player_name, const std::string &password)
+{
```

Comment:
instead of generating all these unconditionally - which is actually a step back - why not create them on get?

the important part is just to delete it all after auth was successful.

Reply:
It will required copy of password in memory.
This goes around it.